src/core/contracts/src/main/scala/Params.scala

```diff
@@ -121,6 +121,18 @@ trait HasOutputCol extends Wrappable {
   def getOutputCol: String = $(outputCol)
 }
 
+trait HasInputCols extends Wrappable {
```

Comment:
This is great thanks but we should be very careful about versioning Core now that we are public. Is there a way to add a `@since` tag or would that be too superfluous?